inttest/common/footloosesuite.go

```diff
@@ -168,10 +169,7 @@ func (s *FootlooseSuite) SetupSuite() {
 
 	ctx, cancel := newSuiteContext(t)
 	var cleanupTasks sync.WaitGroup
-	sctx := suiteCtx{ctx, func() {
-		cancel()
-		cleanupTasks.Wait()
-	}}
+	sctx := suiteCtx{ctx, cancel, cleanupTasks.Wait}
 
 	if !s.ctxPtr.CompareAndSwap(nil, &sctx) {
```

Comment:
side note, but why do we use cas here? I am not sure I understand but when do we run SetupSuite concurrently for the same instance?

Reply:
We could also use `Store` here, since this is currently the only place where the context gets replaced. It's more a defensive measure in case some other code paths get added in the future that'll also write to ctxPtr.

We need to sync the ctxPtr variable since it might be accessed concurrently from multiple goroutines via `s.Context()`.